File: Python/countBattleships.py
class Solution(object):
    def countBattleships(self, board):
        count = 0;
        i = 0
        j = 0

        if(len(board) == 1):
            while(j<len(board[i])):
                if(board[i][j] == 'X'):
                    count = count + 1
                    if ((j+1) < len(board[i]) and board[i][j + 1] == 'X'):
                        j = j + 1
                        while (j + 1 < len(board[i]) and board[i][j + 1] == 'X'):
                            j = j + 1
                j += 1

        else:
            while( i < len(board)):
                j = 0
                while(j < len(board[i])):
                    if(board[i][j] == 'X'):
                        if ((j+1) < len(board[i]) and board[i][j + 1] == 'X'):
                            j = j + 1
                            while (j + 1 < len(board[i]) and board[i][j + 1] == 'X'):
                                j = j + 1
                            count = count + 1
                        elif( i - 1 < 0 or board[i - 1][j] != 'X'):
                            count += 1
                    j += 1
                i += 1

        return count

s = Solution()
print(s.countBattleships(["..","XX"]))

File: C/countBattleships.c
#include <stdlib.h>
#include <stdio.h>

int countBattleships(char** board, int boardRowSize, int boardColSize) {
    int count = 0;
    int i, j;
    for (i = 0; i < boardColSize; i++) {
        for (j = 0; j < boardRowSize; j++) {
            if (board[i][j] == 'X') {
                if ((j + 1) < boardRowSize && board[i][j + 1] == 'X') {
                    j++;
                    while (j + 1 < boardRowSize && board[i][j + 1] == 'X') {
                        j++;
                    }
                    count++;
                } else if(i - 1 < 0 || board[i - 1][j] != 'X') {
                    count++;
                }
            }
        }
    }
    return count;
}

int main(){
    char **board = malloc(sizeof(char*) * 3);

    board[0] = malloc(sizeof(char)*4);
    board[0][0] = 'X';
    board[0][1] = '.';
    board[0][2] = '.';
    board[0][3] = 'X';
    printf("hi\n");


    board[1] = malloc(sizeof(char)*4);
    board[1][0] = '.';
    board[1][1] = '.';
    board[1][2] = '.';
    board[1][3] = 'X';

    board[2] = malloc(sizeof(char)*4);
    board[2][0] = '.';
    board[2][1] = '.';
    board[2][2] = '.';
    board[2][3] = 'X';

    int a = countBattleships(board, 4, 3);

    printf("%d\n",a);
}
//
//scan each line at a time, if horizontal battleship found(2 or more X's just keep going until . is found and them increment the count 
//if vertical battleship is found(only 1 X) check if the square above is a . if it is increment if is a X do nothing
//
